Libraries/LibWeb/WebIDL/Buffers.cpp

```diff
@@ -127,4 +127,31 @@ void ArrayBufferView::write(ReadonlyBytes bytes, u32 starting_offset)
 
 BufferSource::~BufferSource() = default;
 
+// https://webidl.spec.whatwg.org/#buffersource-detached
+bool is_buffer_source_detached(JS::Value const& buffer_source)
+{
+    // A buffer source type instance bufferSource is detached if the following steps return true:
+
+    JS::ArrayBuffer* array_buffer = nullptr;
```

Comment:
Spec calls this variable `jsArrayBuffer`, so we should probably use `js_array_buffer`?